sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -1220,6 +1230,18 @@ struct _pi_ze_event_list_t {
     this->Length = other.Length;
     return *this;
   }
+
+  /// @brief Method to check if this event list contains the event.
+  /// @param Event Event to look for in the wait list.
+  /// @return true if event is found in the event list, false otherwise.
+  bool contains(pi_event Event) {
+    std::vector<pi_event> WaitListVec(PiEventList, PiEventList + Length);
```

Comment:
Why can't you iterate `PiEventList` directly instead of paying the overhead f constructing a vector?

Reply:
I constructed a vector here to be able to use std::find. Do you think it's better just to iterate in a loop till I find the event?

Reply:
> I constructed a vector here to be able to use std::find. Do you think it's better just to iterate in a loop till I find the event?

yes, sure. that's what std::find does anyway

Reply:
Got it, fixed.